app/models/concerns/attachment_jcamp_aasm.rb

```diff
@@ -350,6 +349,21 @@ def delete_edit_peak_after_done
     destroy if %w[edit peak].include?(typname)
   end
 
+  def delete_related_edit_peak(jcamp_att)
+    return unless jcamp_att
+
+    atts = Attachment.where(attachable_id: attachable_id)
+    valid_name = fname_wo_ext(self)
+    atts.each do |att|
+      is_delete = (
+        (att.edited? || att.peaked?) &&
+          att.id != jcamp_att.id &&
```

Comment:
Is that case possible? In combination with line 355 it would mean that an attachment is attached to itsself